### DISCUSSION TITLE
CompletionItem cursor position

### DISCUSSION THREAD

**Discussion by y912765390** [microsoft/monaco-editor#3669] [Q&A]
If I want to set the cursor position, I need to set it“insertTextRules: monaco.languages.CompletionItemInsertTextRule.InsertAsSnippet”
But in this case, the auxiliary input is not triggered unless you press ESC.
I want to input auxiliary. After entering, set the cursor position and normal input can provide input auxiliary. What should I do

**Comment by rcjsuen** [comment]
> If I want to set the cursor position, I need to set it“insertTextRules: monaco.languages.CompletionItemInsertTextRule.InsertAsSnippet” But in this case, the auxiliary input is not triggered unless you press ESC. I want to input auxiliary. After entering, set the cursor position and normal input can provide input auxiliary.

@y912765390 I don't think I understand your question. What is auxiliary input in this context? Can you provide a screenshot or some more details?

**Comment by y912765390** [comment]
monaco.languages.registerCompletionItemProvider('javascript', {
    provideCompletionItems: function (model, position) {
        // find out if we are completing a property in the 'dependencies' object.

        return {
            suggestions: [{
                label: 'myif',
                kind: monaco.languages.CompletionItemKind.Function,
                documentation: 'Describe your library here',
                insertText: 'myif(${1:my-third-party-library}){$2}',
                insertTextRules: monaco.languages.CompletionItemInsertTextRule.InsertAsSnippet,
            }]
        };
    }
});

monaco.editor.create(document.getElementById('container'), {
    value: '',
    language: 'javascript'
});
Sorry, I can't upload my pictures because of the Internet


////////////////
let shouldTips=""
myif(shou){}
////////////////

When I input "my", I will prompt "myif" and press enter to continue to input "Shou". At this time, it should prompt "shoudtips"
I think I can prompt the content without pressing ESC. What should I do

**Comment by rcjsuen** [comment]
@y912765390 Maybe https://github.com/microsoft/monaco-editor/issues/1911#issuecomment-988761965 will help you. That sounds like your issue with snippets?
